ui/imports/shared/status/StatusGifPopup/ConfirmationPopup.qml

```diff
@@ -0,0 +1,88 @@
+import QtQuick 2.13
+import QtQuick.Controls 2.13
+import QtGraphicalEffects 1.0
+
+import StatusQ.Core 0.1
+import StatusQ.Controls 0.1
+
+import utils 1.0
+import shared.panels 1.0
+import shared.stores 1.0
+import shared.controls 1.0
+
+Popup {
+    id: root
+
+    anchors.centerIn: parent
+    height: 278
+    width: 291
+
+    horizontalPadding: 6
+    verticalPadding: 32
+
+    modal: false
+    closePolicy: Popup.NoAutoClose
+
+    background: Rectangle {
+        radius: Style.current.radius
+        color: Style.current.background
+        border.color: Style.current.border
+        layer.enabled: true
+        layer.effect: DropShadow {
+            verticalOffset: 3
+            radius: 8
+            samples: 15
+            fast: true
+            cached: true
+            color: "#22000000"
+        }
+    }
+
+    SVGImage {
+        id: gifImage
+        anchors.horizontalCenter: parent.horizontalCenter
+        anchors.top: parent.top
+        source: Style.svg(`gifs-${Style.current.name}`)
+    }
+
+    StatusBaseText {
+        id: title
+        anchors.horizontalCenter: parent.horizontalCenter
+        anchors.top: gifImage.bottom
+        anchors.topMargin: 8
+        text: qsTr("Enable Tenor GIFs?")
+        font.weight: Font.Medium
+        font.pixelSize: Style.current.primaryTextFontSize
+    }
+
+    StatusBaseText {
+        id: headline
+        anchors.horizontalCenter: parent.horizontalCenter
+        anchors.top: title.bottom
+        anchors.topMargin: 4
+
+        text: qsTr("Once enabled, GIFs posted in the chat may share your metadata with Tenor.")
+        width: parent.width
+        horizontalAlignment: Text.AlignHCenter
+        wrapMode: Text.WordWrap
+        font.pixelSize: 13
+        color: Style.current.secondaryText
+    }
+
+    StatusButton {
+        id: removeBtn
+        objectName: "enableGifsButton"
+        anchors.horizontalCenter: parent.horizontalCenter
+        anchors.bottom: parent.bottom
+        text: qsTr("Enable")
+
+        size: StatusBaseButton.Size.Small
+
+        onClicked: {
+            RootStore.setIsTenorWarningAccepted(true)
+            RootStore.updateWhitelistedUnfurlingSites("media.tenor.com", true)
```

Comment:
We have no reason to keep 2 flag that are doing the exact same thing, we could simply remove setIsTenorWarningAccepted and use the whilelist, indeed the act of whitelisting tenor is approving that we gonna make call

Reply:
`isTenorWarningAccepted` stores that the user confirmed his awareness of privacy implications
`media.tenor.com` is for previewing any link.
This change creates a relationship where the tenor warning will enable also whitelisting for a better UX and disabling the preview will reset the warning. However, you can always enable gifs from others without sending yourself one.

I didn't dive too much in the philosophy and probably a better workflow can be achieved which will need further design and different priorities.